execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -778,12 +778,7 @@ where
                                 return Some(Err(e));
                             }
                         };
-                        debug_assert!({
-                            match &maybe_next_trie {
-                                Some(Trie::Node { .. }) => true,
-                                _ => false,
-                            }
-                        });
+                        debug_assert!({ matches!(&maybe_next_trie, Some(Trie::Node { .. })) });
```

Comment:
Is this a clippy recommend after the changing the toolchain?  I like it either way, but holy seemingly unrelated changes...

Reply:
Yeah.  I'll do a separate PR for the upgrade as Ed suggests.